src/Microsoft.Health.Dicom.Core/Features/Common/IFileStore.cs

```diff
@@ -70,4 +70,14 @@ public interface IFileStore
         VersionedInstanceIdentifier versionedInstanceIdentifier,
         FrameRange range,
         CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Asynchronously deletes a old DICOM file from the file store if the file exists.
+    /// </summary>
+    /// <param name="versionedInstanceIdentifier">The DICOM identifier.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>A task that represents the asynchronous delete operation.</returns>
+    Task DeleteOldFileIfExistsAsync(
```

Comment:
Don't we have some class that represents the format of the remote data? If that's the case, should we not just have `DeleteIfExists` methods that optionally accept these formatters?

Reply:
What do you mean by optionally accepting these formatters. The logic here is unique, it is not just deleting if it exists. It needs to ensure it has copied the new format before deleting the old one